packages/presentation/lib/src/game_bloc/game_bloc.dart

```diff
@@ -0,0 +1,134 @@
+import 'package:domain/domain.dart';
+import 'package:flutter/material.dart';
+import 'package:presentation/src/constants/init_values.dart';
+import 'package:presentation/src/game_bloc/bloc.dart';
+import 'package:presentation/src/game_bloc/bloc_tile.dart';
+import 'package:presentation/src/game_bloc/show_snack_bar_event.dart';
+
+abstract class GameBloc extends Bloc {
+  factory GameBloc(
+    GenerateGuessNumberUseCase generateGuessNumber,
+    MakeAttemptUseCase makeAttempt,
+    GlobalKey<FormState> formKey,
+  ) =>
+      GameBlocImpl(
+        makeAttempt,
+        generateGuessNumber,
+        formKey,
+      );
+
+  Function()? onNewGamePressed();
+
+  Function()? onMakeAttemptPressed();
+
+  void onTextChange(String text);
+
+  String? validate(String? text);
+
+  GlobalKey<FormState> get formKey;
+}
+
+class GameBlocImpl extends BlocImpl implements GameBloc {
+  final GenerateGuessNumberUseCase _generateGuessNumberUseCase;
+  final MakeAttemptUseCase _makeAttemptUseCase;
+  final GlobalKey<FormState> formKey;
+  int _suggestedNumber = blocTileInitValue;
+
+  GameBlocImpl(
+    this._makeAttemptUseCase,
+    this._generateGuessNumberUseCase,
+    this.formKey,
+  );
+
+  @override
+  void initState() {
+    super.initState();
+    emit(tile);
+  }
+
+  @override
+  Function()? onMakeAttemptPressed() =>
+      tile.state == BlocTileState.gameInProgress ||
+              tile.state == BlocTileState.newGame
+          ? _makeAttempt
+          : null;
```

Comment:
imho, no need in moving this logic to bloc. condition, which handles is button disabled or no can be kept in widget. or can be placed to state class in separate method. and in widget u will have the following construction:

`state.isBtnEnabled() ? bloc.makeAttempt : null`